Makefile

```diff
@@ -1,3 +1,5 @@
+# TODO: FIX THIS!! Use NX commands
```

Comment:
Couldn't we have one Makefile per package with the specific needs of each module/package?

Reply:
With NPM workspaces all commands should be ran from the root folder, that's why I kept the Makefile here.
If we opt to use NX, it will also run commands from the root folder. 

ESLint and TS are installed centrally as well (and configured differently in each package, if needed), so they need to run from the root.

So, in general, we'll be running most things from the root folder.

What do you think?

Reply:
I don't like the approach of running commands from root. I am creating another monorepo here and decided to move to Yarn workspaces just because NPM requires to run from root... I like the idea of hiding information in the context you are working, in this case, represented by the folder of the service... and about sharing ts or js configs I don't know if it's good also, specially because the plugin might need some tweaks that we don't necessarily need in monolint "core"

Reply:
IMHO linting/formatting/etc are part of the "workspace standards" so they should be configured once in the root and also run from the root as well most of the times. (since you shouldn't be affected by the other packages... if they were merged, they should be already fine) 
But of course, this requires better tooling than just plain `npm` (nx for the rescue 😄).

I know packages might differ sometimes, but that's why we (can) have "extends" configurations for most of the aforementioned "tooling".

Having a shared `tsconfig` (for linting purposes) in a monorepo is actually part of the [typescript-eslint guide](https://typescript-eslint.io/docs/linting/typed-linting/monorepos) and I was also inspired by Storybook's monorepo.
So, mainly what this means is:
```
/
  packages/
    package-name/
      .estlinrc.js -> optional, only if tweaks are needed
      tsconfig.json -> extends: ../../tsconfig.json + tweaks
  .eslintrc.js
  tsconfig.json
  tsconfig.eslint.json
```

We could switch to `yarn` to have this flexibility of "scoping" the "working-directory" as well. That is fine for me.
But I think for most CI runs that we do we'll run things from the root with e.g. `nx affected` type of commands. To build/test/publish the package(s). Are we not?

---

I have a few questions that are a bit broader than "run or not run commands from the root". (that could be transformed into an issue by itself)

1. How are we planning to keep the versions for the packages?
  Storybook/Jest = they bump all packages inside the monorepo, although that package might not have changes.
  ESlint = not a monorepo, extension and "core" have different dependencies. (which make sense for us as well, the extension should run independently of the "core" version, most of the times)
2. Are we planning to add more packages?

Reply:
Btw, I want to narrow down the scope of this PR to the bare minimum so we can have the monorepo setup up and running ASAP :) 

So I believe next steps are:
- Switch to yarn
- Makefiles for each package

Then we create the issue to discuss the fundamentals from the previous comment :) 

What do you think?